brightwind/load/load.py

```diff
@@ -1357,70 +1361,35 @@ def get_data(measurement_station_uuid, date_from=None, date_to=None):
-                url_end="/resource-data",
-                params={"measurement_location_uuid": measurement_station_uuid,
-                        "date_from": _date_from, "date_to": _date_to})
-            pre_signed_url = response.json()["url"]
-            response = requests.get(pre_signed_url)
-            return response.text
-
-        master_df = pd.DataFrame()
-        with concurrent.futures.ThreadPoolExecutor() as executor:  # optimally defined number of threads
-            res = [executor.submit(get_chunk, chunk[0], chunk[1]) for chunk in chunks]
-            concurrent.futures.wait(res)
-            for result in res:
-                chunk_df = pd.read_csv(StringIO(result.result()),)
-                chunk_df["Timestamp"] = pd.to_datetime(chunk_df["Timestamp"])
-                chunk_df.index = pd.DatetimeIndex(chunk_df.pop("Timestamp"))
-                if master_df.empty:
-                    master_df = chunk_df
+        response = LoadBrightHub.__get_timeseries_data(measurement_station_uuid, date_from, date_to)
+        response_json = response.json()
+
+        assembly_in_progress_err_msg = 'The server is currently busy assembling the timeseries file.'
+        if response.status_code == 503 and assembly_in_progress_err_msg in response_json.get('details'):
+            retry_after = int(response.headers.get('Retry-After', 60))  # Default retry after default 60 seconds
+            for _ in range(4):  # attempt 4 more times
+                time.sleep(retry_after)  # Wait before retrying
+                response = LoadBrightHub.__get_timeseries_data(measurement_station_uuid, date_from, date_to)
+                response_json = response.json()
+                if response.status_code == 503 and assembly_in_progress_err_msg in response_json.get('details'):
+                    retry_after *= 2  # double the retry time for each attempt i.e. back-off.
+                    continue  # try again
                 else:
-                    master_df = pd.concat([master_df, chunk_df])
-        return master_df
+                    # either the call was a success or there was a different error
+                    break  # exit for loop
+            # raise the 503 error if all the attempts have been used up
+            if response.status_code == 503 and assembly_in_progress_err_msg in response_json.get('details'):
+                raise ValueError("{}. {}".format(response_json['error'], response_json['details']))
+
+        if response.status_code == 404 and 'details' in response_json:  # other BrightHub specific errors.
+            raise ValueError("{}. {}".format(response_json.get('error', ''), response_json['details']))
+        elif 'error' in response_json:
+            raise ValueError("Unexpected error: Status Code {}. {}".format(response.status_code, response.text))
+
+        pre_signed_url = response_json["url"]
+        timeseries_response = requests.get(pre_signed_url)
```

Comment:
Consider error handling presigned_url request.

But the only one that comes to mind is if the presigned_url has expired but that should not happen when running this code. Or of there is an internet error